rcgen/src/key_pair.rs

```diff
@@ -29,30 +30,23 @@ use crate::{sign_algo::SignatureAlgorithm, Error};
 
 /// A key pair variant
 #[allow(clippy::large_enum_variant)]
+#[cfg(feature = "crypto")]
 pub(crate) enum KeyPairKind {
 	/// A Ecdsa key pair
-	#[cfg(feature = "crypto")]
 	Ec(EcdsaKeyPair),
 	/// A Ed25519 key pair
-	#[cfg(feature = "crypto")]
 	Ed(Ed25519KeyPair),
 	/// A RSA key pair
-	#[cfg(feature = "crypto")]
 	Rsa(RsaKeyPair, &'static dyn RsaEncoding),
-	/// A remote key pair
-	Remote(Box<dyn SigningKey + Send + Sync>),
```

Comment:
With this removal can we also delete [`Error::RemoteKeyError`](https://github.com/rustls/rcgen/blob/1cae5cdac352aec5bd50e1d83a38a3433ebafd79/rcgen/src/error.rs#L37-L38)?

It doesn't look like it's used anywhere.

Reply:
Alternatively, if the intention is to leave this as a sort of catch-all `Error` that can be used by an external `SigningKey` impl for its `sign()` fn maybe we should make it capable of carrying some kind of context message, or rename it to `SigningError`?

Reply:
It wasn't used before, so these questions are mostly independent of the changes in this PR. I don't disagree but let's discuss this separately?

Reply:
Reasonable :+1: I'll resolve this thread.